### ISSUE TITLE
Check whether the model classes exists

### ISSUE THREAD

**Issue by pyrech** [sonata-project/SonataMediaBundle#818]
Hi,

In several sonata bundles (at least SonataMedia and SonataClassification), the config need to specify model classes with the `class` parameter.

For example (in config.yml):
```yaml

sonata_media:
    # if you don't use default namespace configuration
    class:
        media:             AppBundle\Entity\Media\Media
        gallery:           AppBundle\Entity\Media\Gallery
        gallery_has_media: AppBundle\Entity\Media\GalleryHasMedia
        category:          AppBundle\Entity\Classification\Category
```

However, if a specified class doesn't exist (f.e. typo or wrong copy/paste), some wtf problems can happen anywhere in the application (because some metadata are wrong and then some relations / tables are not created) without obvious reasons.

I propose to add a check on `DependencyInjection\Configuration` to ensure the specified classes exist. Do you agree with this? If yes I could eventually make a PR for all the bundles using the `class` configuration

**Comment by sfavot** [comment]
:+1:

**Comment by Soullivaneuh** [comment]
Looks like a good idea. Could you propose a PR about that?

**Comment by pyrech** [comment]
Yep, I will have a look at it this evening ;)

**Comment by Soullivaneuh** [comment]
Also could you please paste some sample errors when specified class does not exist?

With it, I could evaluate the wtf level of these. ;-)

**Comment by pyrech** [comment]
Sure :) Using sonata media 0c6103251c8e41b0b72f4f885533d8d54a5b3ef2, I found these errors:

When clicking on the btn_add button from an entity edit form
> Impossible to invoke a method ("id") on a null variable in "SonataDoctrineORMAdminBundle:Form:form_admin_fields.html.twig" at line 59

When clicking on the btn_list from an entity edit form or trying to display the admin media listing
> Notice: Undefined index: category in /path/vendor/sonata-project/doctrine-orm-admin-bundle/Guesser/FilterTypeGuesser.php on line 64

**Comment by Soullivaneuh** [comment]
And those errors are thrown because specified class simply not exists?

Indeed, this is not very clear. :-)

**Comment by pyrech** [comment]
Yes :-)

When trying to find the problem, I noticed that the doctrine metadata name use className of Base{Model} (not even my wrong class).

So in `Sonata\EasyExtendsBundle\Mapper\DoctrineORMMapper`, all these checks obviously fail:
```php
if (!array_key_exists($metadata->name, $this->associations)) {
    return;
}
```

So some columns and the tables are not even created. The errors are then quite logical :smile: 
- In the template, it's probably an access to the id of a relation not created (Couldn't find the template part related because the mentionned line 59 is about money_widget...)
- The MediaAdmin declares a `category` with null type. Because the TypeGuesser use metadata to determine the type, it's trying to look for the `category` information which doesn't exist

**Comment by Soullivaneuh** [comment]
I see.

I'm waiting your proposition so. ;-)

**Comment by pyrech** [comment]
As it's a simple fix, should I maker a PR against sonata media 2.3 or against master?

**Comment by Soullivaneuh** [comment]
As branches are mergeable on this project, 2.3 should be a good choice.

Sullivan SENECHAL

**Comment by pyrech** [comment]
Here it is: #820 :)

If you valid this PR, others PRs should be made to:
- sonata-media 2.4 to add the check on category
- sonata-classification 2.2
- sonata-classification master (context was added)
- sonata-page 2.3
- sonata-news 2.2
- sonata-news master (looks like tag and category where changed to tag and collection from sonata-classification)
- sonata-user 2.2
- sonata-notification 2.2
- sonata-comment master
- sonata-timeline master

**Comment by Soullivaneuh** [comment]
> If you valid this PR, others PRs should be made to:

Did you plan it to take a look on them if the first PR is accepted?

**Comment by pyrech** [comment]
Yep, will try to do my best to do it quickly ;)

**Comment by rande** [comment]
@pyrech missing classes cannot be check during the container configuration as easy extends need to container to generate classes. Also the check cannot be done at runtime as this will add extra overhead.

The solution will be to look into doctrine command and add a bit of documentation to make the DX more pleasant.

**Comment by OskarStark** [comment]
closes as #820 is closed, too
